### ISSUE TITLE
Linting not working. Nothing is showing up when there are supposed to be errors.

### ISSUE THREAD

**Issue by adriandelgg** [juanfranblanco/vscode-solidity#274]
Hello, I'm using the Solidity extension & before it used to work just fine and would show linting for errors, but now I never get them. It's clear that the errors are there because when I compile with Hardhat/Truffle it complains, but I'm not getting linting as I'm coding like how Remix IDE does.

I don't understand why it's not working anymore and it's really upsetting since I rely on the linting for work.

**Comment by juanfranblanco** [comment]
Hi @adriandelgg what are your settings?

**Comment by adriandelgg** [comment]
Hey @juanfranblanco, thanks for replying and making this awesome VS Code extension.
I've attached the extension settings I have. It used to work where I would get linting while typing like how Remix does, but now I don't anymore and I have no idea why.
![2021-08-02_10-55_1](https://user-images.githubusercontent.com/32179921/127903771-79fc208d-6a85-4065-b8c6-21d91d3afc82.png)
![2021-08-02_10-55](https://user-images.githubusercontent.com/32179921/127903774-d829395e-c069-4556-a68b-daa31a8ac82c.png)

**Comment by adriandelgg** [comment]
Any ideas as to why it's not working? 

Is there any way I can completely delete the extension and any past versions so that I can have a brand new fresh installation of it?

**Comment by AbhiGene** [comment]
bump on this thread, I am having the same error and it is causing me great trouble

**Comment by juanfranblanco** [comment]
@adriandelgg @ClockworkRainbow apologies this thread went out of track, are you still having the issues?

**Comment by adriandelgg** [comment]
> @adriandelgg @ClockworkRainbow apologies this thread went out of track, are you still having the issues?

It's okay. I'm not sure if I'm supposed to get the exact linting i get like i do with Remix IDE. 

For example, I'm Remix, it detects on the IDE if I can turn a function into view, while on VSCode it only suggests that when I run the compiler. 

Is this the right behaviour?

**Comment by juanfranblanco** [comment]
@adriandelgg does it look like this? This is mainly driven by the compiler in the background.
![error_suggestions_visibility_compiler](https://user-images.githubusercontent.com/562371/135123586-9c57bbf1-e2c8-4c16-85d0-27f0e3672edf.gif)

**Comment by juanfranblanco** [comment]
No you don't need to install anything else.
What I am seeing is that you have enabled the error reporting as you type, you are using the latest version of solidity (which is downloaded, "remote"), you can compile as you have said and it shows the errors. So there might be an issue when starting the server. Can you try the latest version? This might solve your problem.

**Comment by adriandelgg** [comment]
I tried & it's still not working, and now I'm getting a new error.
`Source "@openzeppelin/contracts/access/Ownable.sol" not found: File import callback not supported`

I'm not sure why I keep having issues with the extension.

**Comment by adriandelgg** [comment]
These are my `settings.json` in VS Code:
```json
{
	"telemetry.enableCrashReporter": false,
	"liveServer.settings.donotShowInfoMsg": true,
	"git.autofetch": true,
	"workbench.iconTheme": "material-icon-theme",
	"editor.fontFamily": "'Fira Code', monospace",
	"telemetry.enableTelemetry": false,
	"workbench.startupEditor": "newUntitledFile",
	"editor.renderWhitespace": "none",
	"workbench.editor.decorations.colors": true,
	"oneDarkPro.bold": true,
	"git.enableSmartCommit": true,
	"gitlens.advanced.messages": {
		"suppressGitMissingWarning": true
	},
	"explorer.confirmDragAndDrop": false,
	"[json]": {
		"editor.quickSuggestions": {
			"strings": true
		},
		"editor.suggest.insertMode": "replace",
		"gitlens.codeLens.scopes": ["document"],
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"editor.fontLigatures": "'ss01', 'ss02', 'ss03', 'ss04', 'ss05', 'ss06', 'zero', 'onum'",
	"editor.fontSize": 15.95,
	"javascript.updateImportsOnFileMove.enabled": "always",
	"editor.tabSize": 2,
	"[javascript]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"[typescript]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"[typescriptreact]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"prettier.singleQuote": true,
	"prettier.arrowParens": "avoid",
	"prettier.useTabs": true,
	"prettier.trailingComma": "none",
	"explorer.confirmDelete": false,
	"[javascriptreact]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"[html]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"editor.minimap.renderCharacters": false,
	"editor.cursorSmoothCaretAnimation": true,
	"editor.cursorWidth": 2,
	"editor.tokenColorCustomizations": {
		"textMateRules": [
			{
				"scope": ["entity.name.function", "entity.name.method"],
				"settings": {
					"fontStyle": "bold"
				}
			}
		]
	},
	"editor.semanticHighlighting.enabled": true
	"editor.cursorBlinking": "expand",
	"terminal.integrated.cursorBlinking": true,
	"editor.linkedEditing": true,
	"launch": {
		"configurations": [],
		"compounds": []
	},
	"typescript.updateImportsOnFileMove.enabled": "always",
	"emmet.showSuggestionsAsSnippets": true,
	"emmet.includeLanguages": {
		"javascript": "javascriptreact",
		"typescript": "typescriptreact"
	},
	"solidity.compileUsingRemoteVersion": "v0.8.7+commit.e28d00a7",
	"git.confirmSync": false,
	"tailwindCSS.emmetCompletions": true,
	"files.trimTrailingWhitespace": true,
	"codesnap.backgroundColor": "#fb6f7d",
	"codesnap.boxShadow": "rgba(0, 0, 0, 0.55) 0px 10px 20px",
	"git.enableCommitSigning": true,
	"editor.minimap.enabled": false,
	"editor.formatOnSave": true,
	"[markdown]": {
		"editor.defaultFormatter": "yzhang.markdown-all-in-one"
	},
	"security.workspace.trust.untrustedFiles": "open",
	"terminal.integrated.defaultProfile.linux": "zsh",
	"[jsonc]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"python.pythonPath": "C:\\Users\\adria\\AppData\\Local\\Programs\\Python\\Python39\\python.exe",
	"editor.bracketPairColorization.enabled": true,
	"redhat.telemetry.enabled": false,
	"vs-kubernetes": {
		"vscode-kubernetes.kubectl-path.linux": "/home/CodeOnBlocks/.vs-kubernetes/tools/kubectl/kubectl",
		"vscode-kubernetes.helm-path.linux": "/home/CodeOnBlocks/.vs-kubernetes/tools/helm/linux-amd64/helm",
		"vscode-kubernetes.minikube-path.linux": "/home/CodeOnBlocks/.vs-kubernetes/tools/minikube/linux-amd64/minikube"
	},
	"workbench.colorTheme": "Electronic Moonlight"
}
```

**Comment by adriandelgg** [comment]
I FIGURED OUT WHY IT'S NOT WORKING. The `"solidity.packageDefaultDependenciesDirectory":` needs to be set to my project.

However, how can I make it so it auto detects it? In the screenshot, you can see that in my workspace I have multiple folders with different projects. I had to set the rule for this specific project as `"solidity.packageDefaultDependenciesDirectory": "bp-nft-meetings/hardhat/node_modules"` in my global `settings.json` file, but obviously this isn't ideal because I have multiple projects & don't want to be having to manually change it every time.

![2021-09-28_20-04](https://user-images.githubusercontent.com/32179921/135196159-3d53dc72-e261-45c3-b734-b7da8633ca6b.png)

Is there a way to set the `settings.json` for each individual project or something?

**Comment by juanfranblanco** [comment]
Ah yeah at the moment the settings in a workspace apply to all folders, (workspace settings), the only solution is to open each folder separate. This is the same for the new remappings. So mainly your bp-projects, you will need to open each folder separate, we are all discussing to have a project file, which will make things much easier (eventually)

**Comment by adriandelgg** [comment]
> Ah yeah at the moment the settings in a workspace apply to all folders, (workspace settings), the only solution is to open each folder separately. This is the same for the new remappings. So mainly your bp-projects, you will need to open each folder separate, we are all discussing having a project file, which will make things much easier (eventually)

That'd be awesome. I've noticed that even though the path is set to another project, if I go into a new project it will actually still work, so that's good. But yes, it'd be great to have it detect it all on its own. I'm assuming it works because it finds the necessary info in the other project, but if I import for example `@openzepplin` I'm assuming it may be referencing the `node_modules` from the other project and not the current one I'm on.

**Comment by juanfranblanco** [comment]
@adriandelgg yes that is the side effect, this workspaces folders is rather painful, as you will need folder settings per folder in workspace, but that is not possible.

**Comment by juanfranblanco** [comment]
@adriandelgg try to add a remappings.txt file in the root folder of each project. Here you can map specific files or folder.
ie 

```
@openzeppelin/=lib/openzeppelin-contracts/
ds-test/=lib/ds-test/src/

```
